pytorch_lightning/trainer/trainer.py

```diff
@@ -342,6 +342,12 @@ def __init__(
         super().__init__()
         Trainer._log_api_event("init")
         self.state = TrainerState()
+
+        if distributed_backend is not None:
```

Comment:
the other trainer arg deprecations are handled inside the connectors (see truncated_bptt for example)
I would like to suggest moving this warning to the acc connector, what do you think?

Reply:
@awaelchli Tbh, I was thinking the same. But we do `distributed_backend = distributed_backend or accelerator`, and pass either one of them to the Accelerator connector.

yes, let's do that. I will also pass `training_type` flag to the connector as well. It would be cleaner.

Reply:
Updated.